### PR TITLE
Feature/rv compute

### PR DESCRIPTION
**Pull request by F-WRunTime**
**Description**

Minor changes to CircleCI configuration to support running symbolic analysis with KaaS @ Runtime Verification Inc. 
Minor changes to run-kontrol.sh to improve log aggregation and publishing. 

**Tests**
Lists of tests was increased to full suite of tests. 

**Additional context**

Compute will be offloaded to RV. The burden and resource necessity is handled with remote storage and compute by a service called KaaS. Improvements are provided by RV to the computation environment and caching of proofs to decrease execution time of proofs for Optimism.

### CODE REVIEW

**Comment by Inphi** [comment]
Sorry for responding to this much later.
I'm not too familiar with the kontrol setup (still getting there). @mds1 could you please look at this PR?

File: .circleci/config.yml
Comment (mds1) [inline .circleci/config.yml]: Would be nice to add some comments documenting this new way proofs are run, e.g. what we are doing, any security assumptions/tradeoffs, etc

File: .circleci/config.yml
Comment (mds1) [inline .circleci/config.yml:1451]: Are test results and artifacts are passed to CircleCI on completion, or do they need to be fetched from the github workflow job? This would be another good thing to document in the comments.

Additionally, we never got `store_test_results` working properly (to see individual test runtimes and insights) and I'm unsure why—will this PR resolve that?

File: packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: The `Temporarily unexecuted tests` comment above is now outdated. Also, any reason we don't also run the OptimismPortal2Kontrol tests here?

File: packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh]: nit, first comment line is very short
```suggestion
# workers is the minimum between max_workers and the length of test_list unless
# no test arguments are provided, in which case we default to max_workers
```

File: packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh]: Are these error strings the only thing logged, or do we also get e.g. error messages explaining the failures? I'm assuming these would still be present in the terminal as part of the regular kontrol logs, but the `Handle failure` comment sounds like it might be a TODO?

File: .circleci/config.yml
Comment (F-WRunTime) [reply .circleci/config.yml]: Where would you like this documented? Happy to put something together, but make sure they're visible to the developers who need to see it.

File: .circleci/config.yml
Comment (F-WRunTime) [reply .circleci/config.yml:1451]: I'm happy to work with @raffaele-oplabs on it. However I don't have enough visibility into circleCI to debug it outright. 

Moving to this new CI flow, more than likely a post-op workflow is required to fetch the results to store them into CircleCI.

File: packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh
Comment (F-WRunTime) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: This is likely just due to me being outdated on what is currently the desired test suite. I have ping'd @JuanCoRo on slack and we'll make sure everything is running that should be. But we'll converse further there to resolve the full expected test suite.

File: packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh
Comment (F-WRunTime) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh]: Bad placement of the comment, should've been moved up above / removed. These catches are to make sure we capture failures and clean-up the env properly so we fetch all the appropriate files in the container before shutting down. No further planned TODO here as far as I'm aware.

File: .circleci/config.yml
Comment (mds1) [reply .circleci/config.yml]: Maybe we add the docs to `packages/contracts-bedrock/test/kontrol/README.md` and then just reference it with a comment in this file?

Comment (F-WRunTime) [reply .circleci/config.yml]: Okay sounds good. I will work on this tomorrow after talking with @raffaele-oplabs

File: packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh
Comment (JuanCoRo) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: I think we can execute them, but atm there are two reasons why they're left out:
1. Deployment context: If I understand correctly, there are two different deployment configurations, one for each OptimismPortal version. However, both deployments are written to the `Kontrol-31337.json`, which is used to run the proofs. So, if we want to run both versions, we'll need two different JSON files to account for the different deployments.
2. CPU availability: @F-WRunTime can correct me on this, but I think our machine has 16 cores, which means that we can run at most 16 proofs in parallel. Since 12 out of the 16 proofs are OptimismPortal proofs, running the OptimismPortal2Kontrol tests here would mean running a total of 28 proofs in parallel, which I think our infrastructure doesn't support?
I might have to do some research on this

Comment (JuanCoRo) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: Just updated the Temporarily unexecuted test list btw :ok_hand:

Comment (mds1) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: Eventually OptimismPortal and OptimismPortal2 will be merged into one contract, but until then running both is ideal. If we need to choose just one of them for simplicity in this PR, OptimismPortal2 is preferable since that is what we currently use in production

Comment (F-WRunTime) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: We have 16 cores which is 32 threads. I believe that should be fine. Do you have a rough spec of Mem requirements that may be the limiting factor @JuanCoRo

Comment (JuanCoRo) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: For completeness: on my machine, with the 28 proofs running in parallel, the peak memory usage was 108GB

Comment (JuanCoRo) [reply packages/contracts-bedrock/test/kontrol/scripts/run-kontrol.sh:134]: For completeness:

This has been resolved by *not* including the execution of the Portal2 proofs in this PR. In a follow-up PR, we'll switch to *only* executing the Portal2 proofs instead of the current Portal proofs in a PR that will replace the `StateDiff` cheatcodes in favor of the `vm.dumpState` cheatcode

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]: Can we tweak the phrasing a bit to make it sound less like RV docs and more specific to our use case, e.g. "we have a Runtime Verification assigned token" :)

> Additionally, if you want to receive status updates on the pass/fail of compute executions, you must provide a Fine-Grained token with the necessary statuses permissions to RV.

What does this part mean? What are "compute executions" compared to the proof results?

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]:
```suggestion
A workflow in CircleCI is configured within [.circleci/config.yml](../../../../.circleci/config.yml) and calls on the Github API to execute in the KaaS compute ecosystem. CircleCI is configured with the RV provided Secrets to execute compute jobs in KaaS.
```

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]: Can we expand on how to reproduce CI more? e.g. do we need to pass a certain flag to `pnpm test:kontrol`? And what about getting the tokens locally?

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]: Is this correct?
```suggestion
- `kontrol_prove_report.xml` -- This is a Kontrol Generated JUnit XML Report containing results and performance data.
```

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]:
```suggestion
- `Kontrol Results Folder.zip` -- This contains artifacts produced by kontrol during execution that can be used for debugging. This is expected on pass or failure of a proof run.
```

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]:
```suggestion
To Fetch the artifacts of the execution from KaaS, you can either manually navigate to the summary page using the 'Details' link. Or you can use the GitHub API to pull the artifacts.
```

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (mds1) [inline packages/contracts-bedrock/test/kontrol/README.md]:
```suggestion
Method 1: GitHub's `gh` CLI tool
```

Comment (mds1) [reply packages/contracts-bedrock/test/kontrol/README.md]: Also this is great, super easy way to fetch artifacts 👌

File: packages/contracts-bedrock/test/kontrol/README.md
Comment (F-WRunTime) [reply packages/contracts-bedrock/test/kontrol/README.md]: That is all that has to be done actually. Unless you want to dive into using the run-kontrol.sh script itself to reproduce the CI execution the provided command is that is necessary. I can reference above the documentation related to run-kontrol but that information is well laid out already.

Comment (mds1) [reply packages/contracts-bedrock/test/kontrol/README.md]: Sorry let me clarify: I know `pnpm test:kontrol` is how you run the kontrol tests locally, and it can take args for local/dev/docker mode. But how do you reproduce CI tests that offload compute using `pnpm test:kontrol`? This:

> Running `pnpm run test:kontrol` can run a similar environment locally, allowing for reproducing CI tests.

implies that that it's possible

Comment (F-WRunTime) [reply packages/contracts-bedrock/test/kontrol/README.md]: I see. I mentioned this further down in the document which is the 'curl' command used to run it on the remote compute. Requiring the necessary parameters and tokens to be substituted . I can reword this or maybe best to remove this piece all together.

Comment (mds1) [reply packages/contracts-bedrock/test/kontrol/README.md]: Yea I think it's ok to just delete this sentence and rely on the content further down

**Review by mds1 — APPROVED** [review]
Just needs review from @raffaele-oplabs before merging

**Comment by mds1** [comment]
Once this is merged, we will open a new PR that causes kontrol to fail, and manually dispatch the kontrol proofs. This should trigger an alert to slack if everything is setup correctly, in which case we close that PR.

We must test this way because, as a security measure by GitHub, the new workflow must first be merged to develop before github actions will allow it to trigger

**Review by sebastianst — APPROVED** [review]
CI changes LGTM

**Pull request merged**
